Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -629,7 +670,18 @@ public class DependencyResolver<
     /// - Returns: A satisfying assignment of containers and their version binding.
     /// - Throws: DependencyResolverError, or errors from the underlying package provider.
     public func resolve(constraints: [Constraint]) throws -> [(container: Identifier, binding: BoundVersion)] {
-        return try resolveAssignment(constraints: constraints).map{ ($0.identifier, $1) }
+        // Create a set of all the containers referenced by constraints.
+        let allContainers = Set(constraints.map{ $0.identifier })
+        // Get the un-versioned containers.
+        let unversionedContainers = Set(constraints.filter{ $0.requirement == .unversioned }.map{ $0.identifier })
+        // Create assignment for versioned constraints.
+        var assignment = try resolveAssignment(containers: allContainers.subtracting(unversionedContainers), constraints: constraints)
```

Comment:
I would probably go ahead and lift this to be part of the public API, since in practice clients may want to supply a giant blob of constraints, but only ask for a particular container to be resolved -- and they can't necessarily know in advance exactly which they need to pass.